qiime/core/testing/data_layout.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import collections
+import os
+import os.path
+
+###############################################################################
+#
+# int-sequence:
+#
+#     A sequence of integers stored in a single file. Since this is a sequence,
+#     the integers have an order and repetition of elements is allowed.
+#
+###############################################################################
+
+
+def int_sequence_validator(data_dir):
+    raise NotImplementedError()
+
+
+def int_sequence_to_list(data_dir):
+    with open(os.path.join(data_dir, 'ints.txt'), 'r') as fh:
+        view = []
+        for line in fh:
+            view.append(int(line.rstrip('\n')))
+        return view
+
+
+def list_to_int_sequence(view, data_dir):
+    with open(os.path.join(data_dir, 'ints.txt'), 'w') as fh:
```

Comment:
That's a good idea, right now the assumption is that readers and writers open files in universal-newline mode so `\n` can safely be used, but that assumption could easily be violated and things might not stay cross-platform compatible. Are there specific updates you want to see in this code now, or is this more of a general problem to solve post-alpha?